aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java

```diff
@@ -0,0 +1,505 @@
+  }
+
+  String getRandomTableName() {
+    return LF_TEST_TABLE_PREFIX  + UUID.randomUUID().toString().replace("-", "");
+  }
+
+  private static void waitForIamConsistency() throws Exception {
+    Thread.sleep(IAM_PROPAGATION_DELAY); // sleep to make sure IAM up to date
+  }
+
+  private static LakeFormationClient buildLakeFormationClient(String roleArn, String sessionName, String region) {
+    AssumeRoleRequest request = AssumeRoleRequest.builder()
+        .roleArn(roleArn)
+        .roleSessionName(sessionName)
+        .durationSeconds(ASSUME_ROLE_SESSION_DURATION)
+        .build();
+
+    LakeFormationClientBuilder clientBuilder = LakeFormationClient.builder();
+
+    clientBuilder.credentialsProvider(
+        StsAssumeRoleCredentialsProvider.builder()
+            .stsClient(StsClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build())
+            .refreshRequest(request)
+            .build());
+
+    clientBuilder.region(Region.of(region));
+    clientBuilder.httpClientBuilder(UrlConnectionHttpClient.builder());
+    return clientBuilder.build();
+  }
+
+  private static GlueClient buildGlueClient(String roleArn, String sessionName, String region) {
+    AssumeRoleRequest request = AssumeRoleRequest.builder()
+        .roleArn(roleArn)
+        .roleSessionName(sessionName)
+        .durationSeconds(ASSUME_ROLE_SESSION_DURATION)
+        .build();
+
+    GlueClientBuilder clientBuilder = GlueClient.builder();
+
+    clientBuilder.credentialsProvider(
+        StsAssumeRoleCredentialsProvider.builder()
+            .stsClient(StsClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build())
+            .refreshRequest(request)
+            .build());
+
+    clientBuilder.region(Region.of(region));
+    clientBuilder.httpClientBuilder(UrlConnectionHttpClient.builder());
+    return clientBuilder.build();
+  }
+
+  private static void registerResource(String s3Location) {
+    String arn = getArnForS3Location(s3Location);
+    try {
+      lf.registerResource(RegisterResourceRequest.builder()
+          .resourceArn(arn)
+          .roleArn(lfRegisterPathRoleArn)
+          .useServiceLinkedRole(false).build());
+      // when a resource is registered, LF will update SLR with necessary permissions which has a propagation delay
+      waitForIamConsistency();
+    } catch (AlreadyExistsException e) {
+      LOG.warn("Resource {} already registered. Error: {}", arn, e);
+    } catch (Exception e) {
+      // ignore exception
+    }
+  }
+
+  private static void deregisterResource(String s3Location) {
+    String arn = getArnForS3Location(s3Location);
+    try {
+      lf.deregisterResource(DeregisterResourceRequest.builder().resourceArn(arn).build());
+    } catch (EntityNotFoundException e) {
+      LOG.info("Resource {} not found. Error: {}", arn, e);
+    }
+  }
+
+  private static String createPolicyArn(String policyName) {
+    return TEST_POLICY_ARN
```

Comment:
I don't think we need to do string replace, we can just do `String.format("arn:%s:iam::%s:policy/%s", ...)` here